addon/@ember/test-waiters/wait-for.ts

```diff
@@ -96,3 +138,54 @@ function isThenable(
     typeof maybePromise.then === 'function'
   );
 }
+
+function isGenerator(
+  maybeGenerator: any | CoroutineGenerator<unknown>
+): maybeGenerator is CoroutineGenerator<unknown> {
+  // Because we don't have Symbol.iterator in IE11
+  return (
+    typeof maybeGenerator.next === 'function' &&
+    typeof maybeGenerator.return === 'function' &&
+    typeof maybeGenerator.throw === 'function'
+  );
+}
+
+const GENERATOR_WAITER = buildWaiter('@ember/test-waiters:generator-waiter');
+
+function waitForGenerator<T>(
+  generator: CoroutineGenerator<T>,
+  label?: string
+): CoroutineGenerator<T> {
+  GENERATOR_WAITER.beginAsync(generator, label);
+
+  let isWaiting = true;
+  function stopWaiting() {
+    if (isWaiting) {
+      GENERATOR_WAITER.endAsync(generator);
+      isWaiting = false;
+    }
+  }
+
+  return {
+    next(...args) {
+      try {
+        let val = generator.next(...args);
```

Comment:
I _vaguely_ think we need to rewrap in this case, right?

Reply:
We chatted through this and since we are only doing Generators (not the more general iterator) this is fine.